tests/README.md

```diff
@@ -0,0 +1,87 @@
+## HOW TO CREATE A UNIT TEST.
+
+Unit tests should test only small parts of a program. For example,
+a single routine or function.
+
+Source code for a test shall be placed in a program
+specific directory under ./tests.
+
+The name of the unit test source file shall begin with
+"ftst_" (for Fortran) or "tst_" (for C/C++).
+
+A unit test shall have the following components.
+For an example, see [ftst_readnml.F90](filter_topo/ftst_readnml.F90).
+
+- A prologue describing the test and its author.
+- The test should begin with a print to standard
+output, such as "Starting test of ..."
+- A call to the function or routine to be tested.
+- Output from the function or routine should be 
+compared to expected values. If wrong values are
+returned, the test should stop with a bad status.
+- If the test is successful, that message
+should be printed to standard output, i.e., print*,"SUCCESS!"
```

Comment:
The test must return an exit code of 0 for success, and any other value for failure.
The test must never rely on human inspection of results - the test must be fully automated.

Reply:
Updated.